src/Models/HydrostaticFreeSurfaceModels/split_explicit_free_surface_kernels.jl

```diff
@@ -288,44 +284,44 @@ function split_explicit_free_surface_step!(free_surface::SplitExplicitFreeSurfac
 
     grid = free_surface.η.grid
 
-    # we start the time integration of η from the average ηⁿ     
-    Gu  = model.timestepper.G⁻.u
-    Gv  = model.timestepper.G⁻.v
-    Guⁿ = model.timestepper.Gⁿ.u
-    Gvⁿ = model.timestepper.Gⁿ.v
-
-    fill_halo_regions!((free_surface.state.U̅, free_surface.state.V̅))
-    
-    @apply_regionally setup_split_explicit!(free_surface.auxiliary, free_surface.state, free_surface.η, grid, Gu, Gv, Guⁿ, Gvⁿ, χ)
+    # Wait for previous set up
+    wait_free_surface_communication!(free_surface, architecture(grid))
 
-    fill_halo_regions!((free_surface.auxiliary.Gᵁ, free_surface.auxiliary.Gⱽ))
-    
+    # reset free surface averages
     @apply_regionally begin 
+        initialize_free_surface_state!(free_surface.state, free_surface.η)
         # Solve for the free surface at tⁿ⁺¹
         iterate_split_explicit!(free_surface, grid, Δt)
-        # this is the only way in which η̅ is used: as a smoother for the substepped η field
+        # Reset eta for the next timestep
         set!(free_surface.η, free_surface.state.η̅)
+    end
+
+    fields_to_fill = (free_surface.state.U̅, free_surface.state.V̅)
+    fill_halo_regions!(fields_to_fill; async = true)
 
-        # Wait for predictor velocity update step to complete and mask it if immersed boundary.
+    # Preparing velocities for the barotropic correction
+    @apply_regionally begin 
         mask_immersed_field!(model.velocities.u)
         mask_immersed_field!(model.velocities.v)
     end
 
-    fill_halo_regions!(free_surface.η)
-
     return nothing
 end
 
 # Change name
 const FNS = FixedSubstepNumber
 const FTS = FixedTimeStepSize
 
+# since weights can be negative in the first few substeps (as in the default averaging kernel), 
+# we set a minimum number of substeps to execute to avoid numerical issues
+const MINIMUM_SUBSTEPS = 5
```

Comment:
is the convention all caps for consts?

Reply:
That's what we usually use (like in like 312 and 313)

Reply:
cool